src/System.Text.Json/tests/WritableJsonApiTests.cs

```diff
@@ -624,6 +966,324 @@ public static void TestAquiringAllProperties()
             };
 
             IEnumerable<JsonNode> fullTimeEmployees = employees.GetAllProperties("FTE");
+
+            Assert.Equal(3, fullTimeEmployees.Count());
+            Assert.True(fullTimeEmployees.Contains((JsonString)"John Smith"));
+            Assert.True(fullTimeEmployees.Contains((JsonString)"Ann Predictable"));
+            Assert.True(fullTimeEmployees.Contains((JsonString)"Byron Shadow"));
+        }
+
+        /// <summary>
+        /// Transforming JsoneNode to JsonElement
+        /// </summary>
+        [Fact]
+        public static void TestTransformingJsonNodeToJsonElement()
+        {
+            // Send Json through network, should not throw any exceptions:
+            JsonNode employeeDataToSend = EmployeesDatabase.GetNextEmployee().Value;
+            Mailbox.SendEmployeeData(employeeDataToSend.AsJsonElement());
+        }
+
+        /// <summary>
+        /// Transforming JsonElement to JsonNode
+        /// </summary>
+        [Fact]
+        public static void TestTransformingJsonElementToJsonNode()
+        {
+            // Retrieve Json from network, should not throw any exceptions:
+            JsonNode receivedEmployeeData = JsonElement.DeepCopy(Mailbox.RetrieveMutableEmployeeData());
+            if (receivedEmployeeData is JsonObject employee)
+            {
+                employee["name"] = new JsonString("Bob");
```

Comment:
What would happen in here if employee already had a "name"? Would it get added as an extra property? Would we internally just change the value if it already exists? Should we have a method called something like `AddOrUpdatePropertyValue(string propertyName, JsonNode propertyValue)`?

Reply:
I assumed it would update it. It probably depends on if we really want to support not unique keys: https://github.com/kasiabulat/corefx/pull/1#discussion_r305027749